specs/consensus/overview.md

```diff
@@ -411,19 +411,77 @@ correct behaviour provided that `p` is able to prove the existence of a
+processes may have distinct associated [voting powers](#voting-power) in different heights.
+
+#### Determinism
+
+Given a consensus height `h` and the set of processes running height `h`,
+the `proposer(h, r)` function **must be deterministic**.
+This means that any two correct processes that invoke `proposer(h, r)` with the
+same inputs, including the implicit input that is the set of processes running
+consensus height `h` and associated voting powers, should receive the exactly
+same output (process).
+
+#### Correctness
+
+The main goal of the `proposer(h, r)` function is to eventually select a
+correct process to coordinate a round of consensus and to propose an
+appropriate value for it.
+A correct implementation of the function must guarantee that, for every height
+`h`, there is a round `r* >= 0` where `proposer(h, r*)` returns a process `p`
+that is a correct process: `p` does not misbehave nor crash.
+
+Fortunately, it is relatively simple to produce a correct implementation of the
+`proposer(h, r)` method: it is enough to ensure that every process running
+consensus height `h` is selected as the proposer for at least one round.
+In other words, it is enough that processes take turns as the proposers of
+different rounds of a height.
+
+#### Fairness
+
+Tendermint is a consensus algorithm that adheres to the _rotating coordinator_
+approach.
+This means that the role of coordinating rounds and proposing values is
+expected to be played by different processes over time, in successive
+heights of consensus.
+(In contrast to the _fixed coordinator_ approach, where the coordinator or
+proposer is only replaced when it is suspected to be faulty).
+
+While a correct `proposer(h, r)` implementation eventually selects every
+process as the proposer of a round of height `h`, being the proposer of the
+first round of a height is the most relevant role, since most heights of
+consensus are expected to be finalized in round 0.
+A fair proposer selection algorithm should therefore ensure that all processes
+have, over a reasonable long sequence of heights `h`, a similar chance of being
+selected as `proposer(h, 0)`, thus to propose the value that is most likely to be
+decided on that height.
+
+In the case of Proof-of-Stake (PoS) blockchains, where processes are assumed to
+have distinct voting powers, a fair proposer selection algorithm should ensure
+that every process is chosen,  over a reasonable long sequence of heights, as
+`proposer(h, 0)` in a number of heights `h` that is proportional to its voting
+power.
+Observe that when the set of processes running consensus varies over heights,
+so as the voting power associated to each process, producing a fair proposer
+selection algorithm becomes more challenging.
+The [proposer selection procedure of CometBFT][cometbft-proposer], a Tendermint
+implementation in Go, is a useful reference of how fairness can be achieved.
```

Comment:
Should we mention (not explain) that Comet's proposer selection maintains priorities (I guess as part of the validator set) to achieve fairness?

Reply:
This is a good point. Refer to the quoted block that I have added with the reference to Comet and a minimal discussion regarding determinism and internal state of the algorithm.

I am trying to keep implementation details aside of this documentation, that is why I added this quoted block.

Reply:
Thanks for adding the paragraph. I contains just the right level of detail.